File: ComponentSelector.Application/Helpers/QueryParams/ComponentQueryParams.cs
namespace ComponentSelector.Application.Helpers.QueryParams;

public class ComponentQueryParams : PaginationQueryParams
{
    public string? Category { get; set; }
    public string? OrderBy { get; set; } 
    public string? OrderByDesc { get; set; }
    public int? MinPrice { get; set; }
    public int? MaxPrice { get; set; }
    public bool? Availability { get; set; }
    public string? SearchTerm { get; set; }
}

File: ComponentSelector.Client/src/app/_models/queryParams/componentQueryParams.ts
import { PaginationParams } from './paginationParams';

export class ComponentQueryParams extends PaginationParams {
  category?: string;
  orderBy?: string = 'id';
  orderByDesc?: string;
  minPrice?: number;
  maxPrice?: number;
  availability?: boolean;
  searchTerm?: string;
}
